### PR TITLE
feat: connector list filters

### PR DESCRIPTION
**Pull request by rolznz**
Fixes https://github.com/getAlby/bitcoin-connect/issues/11 and https://github.com/getAlby/bitcoin-connect/issues/89

This allows you to provide a `filters="nwc"` argument to the button or modal to only show nwc connectors.

I chose this approach to support future filters (maybe such as realtime payment options for gaming, which might exclude some future mobile wallets)

I think also should be considered how we can allow the user to reorder the connectors (and how to fairly order them by default?)

- [x] Update React wrapper package

### CODE REVIEW

**Comment by bumi** [comment]
hmm, do we really need this? 

is it maybe enough that people can define the list of offered connectors which would directly then also solve the ordering.

`options="injected,albynwc,mutinynwc,other,connector,names"` 
if this is not set we have a default
and then we over this list and render the connectors.

**Comment by reneaaron** [comment]
I generally like @rolznz approach to just define the connection type as an app, so I don't have to update the list of supported wallets as a consumer of bitcoin-connect. 

The point of bitcoin-connect is to abstract away implementation details so I as a developer don't have to care about whether the users chooses e.g. which NWC implementation. I just need to define the features my app needs (e.g. with NWC doing payments while the user is offline, etc)

**Comment by rolznz** [comment]
I had a call with Bumi, we decided to go ahead with the current solution for now, with an alpha flag on the filters attribute as it might change in future versions.

File: src/components/bc-connector-list.ts [hunk 1]
```diff
@@ -10,15 +10,24 @@ import './connectors/index.js';
...
+    connectors.push(html`<bc-nwc-connector></bc-nwc-connector>`);
+    if (!this._filters || this._filters.indexOf('nwc') === -1) {
+      if (window.webln) {
+        connectors.push(
+          html`<bc-extension-connector></bc-extension-connector>`
+        );
+      }
+      connectors.push(html`<bc-lnbits-connector></bc-lnbits-connector>`);
+      connectors.push(html`<bc-lnc-connector></bc-lnc-connector>`);
+    }
```
[the remark this answers is not in the document]
Comment (rolznz) [reply src/components/bc-connector-list.ts:27]: That would be a better internal implementation if we go that way. Or we remove it and just allow to provide a list of connectors as you originally suggested, then there is no complexity here

[the remark this answers is not in the document]
Comment (rolznz) [reply src/components/bc-connector-list.ts:27]: I will merge it for now, let's re-consider it when the next requirement comes

**Pull request merged**